packages/opentelemetry-sdk-trace-web/test/utils.test.ts

```diff
@@ -593,4 +607,172 @@ describe('utils', () => {
       assert.strictEqual(urlObj.pathname, '/%E4%BD%A0%E5%A5%BD');
     });
   });
+
+  describe('getXHRBodyLength', () => {
+    it('should compute body length for Document payload', () => {
+      // webworkers don't have DOMParser
+      if (typeof DOMParser === 'undefined') {
+        assert.ok(true);
+        return;
+      }
+      const doc = new DOMParser().parseFromString(
+        '<html><head><head/><body><p>hello world</p></body>',
+        'text/html'
+      );
+
+      assert.ok(getXHRBodyLength(doc) > 0, 'body length is 0');
+    });
+    it('should compute body length for Blob payload', () => {
+      const blob = new Blob(['hello world'], {
+        type: 'text/plain',
+      });
+
+      assert.strictEqual(getXHRBodyLength(blob), 11);
+    });
+    it('should compute body length for ArrayBuffer/ArrayBufferView payload', () => {
+      const arrayBuffer = new Uint8Array([1, 2, 3]).buffer;
+
+      assert.strictEqual(getXHRBodyLength(arrayBuffer), 3);
+      assert.strictEqual(getXHRBodyLength(new ArrayBuffer(8)), 8);
+      assert.strictEqual(getXHRBodyLength(new ArrayBuffer(8).slice(0, 2)), 2);
+      assert.strictEqual(getXHRBodyLength(new ArrayBuffer(0)), 0);
+    });
+    it('should compute body length for FormData payload', () => {
+      const formData = new FormData();
+      formData.append('key1', 'true');
+      formData.append('key2', 'hello world');
+
+      assert.strictEqual(getXHRBodyLength(formData), 26);
+      assert.strictEqual(getXHRBodyLength(new FormData()), 0);
+    });
+    it('should compute body length for URLSearchParams payload', () => {
+      const search = new URLSearchParams({
+        key1: 'true',
+        key2: 'hello world',
+      });
+
+      assert.strictEqual(getXHRBodyLength(search), 26);
+      assert.strictEqual(getXHRBodyLength(new URLSearchParams()), 0);
+    });
+    it('should compute body length for string payload', () => {
+      const jsonString = JSON.stringify({
+        key1: 'true',
+        key2: 'hello world',
+      });
+      assert.strictEqual(getXHRBodyLength(jsonString), 36);
+      assert.strictEqual(getXHRBodyLength('hello world'), 11);
+      assert.strictEqual(getXHRBodyLength(''), 0);
+    });
+  });
+
+  describe('getFetchBodyLength', () => {
```

Comment:
Would it make sense to have a test(s) for fetch requests which can't have a body (`GET`/`DELETE`) and/or have an empty body?